Modules/_threadmodule.c

```diff
@@ -359,6 +360,8 @@ thread_run(void *boot_raw)
 
     // Wait until the handle is marked as running
     PyEvent_Wait(&boot->handle_ready);
+    // Signal that this Thread is running
+    _PyEvent_Notify(&boot->handle_running);
```

Comment:
Reference to your issue description (**This MemoryError can occur at the C level during the `PyObject_Call` to the `_bootstrap` method**), It might be better to move this statement in the successful block of the call to `PyObject_Call`.